edb/edgeql/qltypes.py

```diff
@@ -172,10 +174,35 @@ def _missing_(cls, name):
         return cls(name.title())
 
 
-class Multiplicity(s_enum.StrEnum):
+class MultiplicityValue(s_enum.StrEnum):
     ZERO = 'ZERO'  # This is valid for empty sets
     ONE = 'ONE'
     MANY = 'MANY'
+    UNKNOWN = 'UNKNOWN'
+
+    def is_one(self) -> bool:
+        return self is MultiplicityValue.ONE
+
+    def is_many(self) -> bool:
+        return self is MultiplicityValue.MANY
+
+    def is_zero(self) -> bool:
+        return self is MultiplicityValue.ZERO
+
+
+@dataclasses.dataclass(frozen=True, eq=False)
```

Comment:
Probably worth adding a quick comment here

Reply:
Actually, I think I'll make the structured multiplicity private to inference.  I originally made it this way to minimize the `multiplicity.py` diff, but it had grown very large anyway.